src/postgres/postgresConnectionStrings.ts

```diff
@@ -0,0 +1,41 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ConnectionOptions, parse } from "pg-connection-string";
+import { postgresDefaultPort } from "../constants";
+import { ParsedConnectionString } from "../ParsedConnectionString";
+import { nonNullProp } from "../utils/nonNull";
+
+export async function parsePostgresConnectionString(connectionString: string): Promise<ParsedPostgresConnectionString> {
+    const config: ConnectionOptions = parse(connectionString.trim());
+    return new ParsedPostgresConnectionString(connectionString, config);
+}
+
+export async function createPostgresConnectionString(host: string, username?: string | undefined, password?: string | undefined): Promise<ParsedPostgresConnectionString> {
+    let connectionString: string;
+    if (username && password) {
+        connectionString = `postgres://${username}:${password}@${host}:${postgresDefaultPort}`;
+    } else {
+        connectionString = `postgres://${host}:${postgresDefaultPort}`;
+    }
+    return await parsePostgresConnectionString(connectionString);
+}
+
+export class ParsedPostgresConnectionString extends ParsedConnectionString {
+    public readonly hostName: string;
+    public readonly username: string;
+    public readonly password: string;
+    public readonly name: string;
+    public readonly port: string;
+
+    constructor(connectionString: string, config: ConnectionOptions) {
+        super(connectionString, config.database?.replace(/[']+/g, ''));
```

Comment:
Connection strings contain quotes around the database name to preserve if full database name (in case it contains extra leading or trailing whitespace). And these quotes are read in as part of the string by parse function from pg-connection-string.

Reply:
Could it use double quotes instead of single quotes? Also, is it possible to have a database name with a single quote in the name?

Reply:
I haven't found any standard online for using double or single quotes. Why would you prefer double quotes? That's a good question. Actually database names can have pretty much any character including  single and double quotes.

Reply:
_I_ don't prefer double quotes. I'm saying the _user_ could use double quotes and then your code would have problems.

Also, if the database name can have quotes in it, that will also cause problems with your current 'replace' logic.

Reply:
That's true. Fixed this with replacing double quotes only and even the copy connection string outputs database name with double quotes around it.